src/core/shape_inference/include/multinomial_shape_inference.hpp

```diff
@@ -43,10 +43,8 @@ std::vector<TRShape> shape_infer(const Multinomial* op,
         } else {
             result_shape = ov::PartialShape::dynamic(1);
         }
+        result_shape.insert(result_shape.begin(), input_shape[0]);
```

Comment:
(Optional change) 
Maybe is possible to move this insert  before check `if (num_samples) {`
and then inside this `if...else` append at end required shape (dimension(s)) by `push_back` or `std::copy`.

Appending new elements in vectors is more efficient when added at end not inserted inside vector (simpler regarding memory allocation and there is no need reorder elements in vector).

Reply:
Idea applied, now it uses push_backs